odoo/addons/base/models/ir_qweb.py

```diff
@@ -224,6 +245,28 @@ def _compile_directive_call_assets(self, el, options, indent):
 
     # method called by computing code
 
+    def _start_profiling(self, directive, attrib, xpath, options):
+        current_thread = threading.current_thread()
+
+        qweb_hooks = getattr(current_thread, 'qweb_hooks', (None,))
+        for qweb_hook in qweb_hooks:
+            qweb_hook('begin', self.env, view_id=options.get('ref'), xpath=xpath, directive=directive, attrib=attrib, arch=options.get('document'))
```

Comment:
Is `options.get('document') ` the complete view?

Reply:
yes it's the xml view